pkg/reconciler/route/domains/domains.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package domains
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/knative/serving/pkg/network"
+	"github.com/knative/serving/pkg/reconciler/route/config"
+)
+
+// HTTPScheme is the string representation of http.
+const HTTPScheme string = "http"
+
+// SubdomainName generates a name which represents the subdomain of a route
+func SubdomainName(r *v1alpha1.Route, suffix string) string {
+	if suffix == "" {
+		return r.Name
+	}
+	return fmt.Sprintf("%s-%s", r.Name, suffix)
```

Comment:
Can we make this a template like the one below?

Reply:
I'm not quite seeing the value of templating the subdomain name right now. Perhaps a separate change later?

Reply:
+1 to separate change.